### PR TITLE
O3-1604: Allow changing Drug Formulation when dispensing

### PR DESCRIPTION
**Pull request by mogoodrich**

### CODE REVIEW

**Comment by mogoodrich** [comment]
Main change in this commit is to factor out the display of a Medication into a new getMedicationDisplay utility method, and to change the display of the non-coded use case to display both the codeable concept and any drugNonCoded value that may have been entered.

File: src/components/history-and-comments.component.tsx [hunk 1]
```diff
@@ -88,7 +88,7 @@ const HistoryAndComments: React.FC<{
                 {formatDatetime(parseDate(dispense.whenHandedOver))}
               </h5>
               <MedicationEventCard
-                medication={dispense}
+                medicationEvent={dispense}
```
Comment (mogoodrich) [inline src/components/history-and-comments.component.tsx:91]: tangential change... I renamed this attribute for clarity

File: src/components/medication-dispense-review.component.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { TextArea, ComboBox, Dropdown, NumberInput } from "@carbon/react";
 import { useLayoutType } from "@openmrs/esm-framework";
 import { useTranslation } from "react-i18next";
 import {
-  getConceptUuidCoding,
+  getConceptCodingUuid,
```
Comment (mogoodrich) [inline src/components/medication-dispense-review.component.tsx:8]: Tangential: slight reworking off the name of the function

File: src/utils.ts [hunk 3]
```diff
@@ -101,11 +101,44 @@ export function getMedicationsByConceptEndpoint(conceptUuid: string) {
...
+
+/**
+ * Given an array of CodeableConcept codings, return the display for the first one without an associated system (which should be the concept-referenced-by-uuid coding)
+ * @param codings
+ */
+export function getConceptCodingDisplay(codings: Coding[]) {
+  return getConceptCoding(codings)?.display;
+}
+
+/**
```
Comment (mogoodrich) [inline src/utils.ts:127]: This new function is the  heart of this PR...  I've extracted the display of a medication out to this method and tweaked the display as described below.

**Review by mseaton — APPROVED** [review]

**Pull request merged**